solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -701,28 +703,20 @@ protected ElevationProvider createElevationProvider(Map<ElevatingQuery, Elevatio
    * Analyzes the provided query string and returns a concatenation of the analyzed tokens.
    */
   public String analyzeQuery(String query) {
-    StringBuilder concatenatedTerms = new StringBuilder();
-    analyzeQuery(query, null, concatenatedTerms);
-    return concatenatedTerms.toString();
+    StringBuilder concatTerms = new StringBuilder();
+    analyzeQuery(query, concatTerms::append);
+    return concatTerms.toString();
   }
 
   /**
-   * Analyzes the provided query string, tokenizes the terms and add them to either the provided {@link Collection} or {@link Appendable}.
-   *
-   * @param queryTerms The {@link Collection} that receives the terms; or null if none.
-   * @param concatenatedTerms The {@link Appendable} that receives the terms; or null if none.
+   * Analyzes the provided query string, tokenizes the terms, and adds them to the provided {@link Consumer}.
    */
-  protected void analyzeQuery(String query, Collection<String> queryTerms, Appendable concatenatedTerms) {
+  protected void analyzeQuery(String query, Consumer<String> termsConsumer) {
     try (TokenStream tokens = queryAnalyzer.tokenStream("", query)) {
       tokens.reset();
       CharTermAttribute termAtt = tokens.addAttribute(CharTermAttribute.class);
       while (tokens.incrementToken()) {
-        if (queryTerms != null) {
-          queryTerms.add(termAtt.toString());
-        }
-        if (concatenatedTerms != null) {
-          concatenatedTerms.append(termAtt);
-        }
+        termsConsumer.accept(termAtt.toString());
```

Comment:
the termAtt.toString() is a bit of a shame since as a CharSequence, the string appending consumer needn't get the String.  But it's not a big deal.

Reply:
I'll change to a Consumer<CharSequence>.